### PR TITLE
Add units for fit parameters to model fitting

### LINKED ISSUE #32453: Model Analysis fit parameters sometimes have wrong units
**Original reporter:** James Lord

**Describe the bug**
Sometimes the wrong units are used for fit parameters in the model fitting tab. This is because it will check the `UnitFactory` for a unit to the parameter name and use that unit if it finds one. This unit isn't always correct however, and so we should avoid doing this.

In the future we will need to get the units of the fit parameters from somewhere, I am open to suggestions.

**To Reproduce**
1. Load MUSR 62260
2. Add function GausOsc
3. Sequentially fit all
4. Go to results tab and create the table
5. Select Phi on the x or y axis. It has units of degrees when it should be radians.

**Expected behavior**
Use no units if plotting a fit parameter because there is no easy way to tell what units to use.

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**
This PR changes Model fitting to try and get the units for its plots from a dictionary of parameters and log names. This is current a limited list but can be expanded easily if a log or parameter is widly used with the same unit.

**Report to:** James Lord

**To test:**

1. Open Muon analysis
2. Load MUSR 62260
3. In the fitting tab add a GausOsc function
4. In Sequential fitting tab click `Sequentialy fit all`
5. In the Results tab check sample_temp and sample_magn_field in Log values
6. Output results table.
7. Go to the model fitting tab
8. Change the selected data in the X axis to different parameters and check that in the plot the units for, sample_temp, sample_magn_field, sigma, Frequency and Phi are correct.
9. Repeat with the y axis.

Fixes #32453 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/contexts/fitting_contexts/model_fitting_context.py [hunk 1]
```diff
@@ -6,6 +6,17 @@
...
 from mantidqtinterfaces.Muon.GUI.Common.contexts.fitting_contexts.basic_fitting_context import BasicFittingContext
 
+PARAMETERUNITS = {
+    'Frequency': ['MHz', 'MHz'],
+    'Lambda': [r'\mu s^{-1}', r'$\mu$ $s^{-1}$'],
+    'Phi': ['Radians', 'Radians'],
+    'sample_magn_field': ['G', 'G'],
+    'sample_temp': ['K', 'K'],
+    'Sigma': [r'\mu s^{-1}', r'$\mu$ $s^{-1}$'],
+    'test1': ['test_unit_1', 'test_unit_1'],
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/contexts/fitting_contexts/model_fitting_context.py:16]: do we want to keep the test values and we should add a comment explaining why there are two versions of all the units

Comment (Harrietbrown) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/contexts/fitting_contexts/model_fitting_context.py:16]: Sure, I put them in for unit tests but I can use any existing unit for that

Comment (AnthonyLim23) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/contexts/fitting_contexts/model_fitting_context.py:16]: the over option is to remove them here and then replace the dict in the unit test with one that includes these

**Review by AnthonyLim23 — APPROVED** [review]
This is a nice change and gives a big quality of life improvement for users

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we please get this merged? I am expecting it to cause some conflicts in another PR that we need working before the end of the week

**Pull request merged**